api/graphql/mutations/network.js

```diff
@@ -2,12 +2,50 @@ import validateNetworkData from '../../models/network/validateNetworkData'
 import underlyingUpdateNetwork from '../../models/network/updateNetwork'
 import convertGraphqlData from './convertGraphqlData'
 
-export function updateNetwork (userId, { id, data }) {
+// TODO: more integrated `isAdmin` handling for mutations?
+export async function networkMutationPermissionCheck ({ userId, isAdmin = false }, networkId) {
+  if (isAdmin) return
+  if (!await NetworkMembership.hasModeratorRole(userId, networkId)) {
+    throw new Error("You don't have permission to modify that network.")
+  }
+}
+
+export async function addCommunityToNetwork (user, { communityId, networkId }) {
+  await networkMutationPermissionCheck(user, networkId)
+  await Community
+    .where('id', communityId)
+    .save('network_id', networkId, { method: 'update', patch: true })
+  return Network.find(networkId, { withRelated: [ 'communities' ] })
```

Comment:
@levity See what you think of the above. I suppose it reads well... it feels a little strange, because at each step I'm relying on the underlying code throwing to stop us reaching statements further on in the function. But it's clean-ish? If permissions fail, we throw... if the db operation fails, we throw... and graphql-tools exposes the error in the response.

Reply:
Looks fine to me!